.github/workflows/cd-frontend.yml

```diff
@@ -28,9 +30,29 @@ on:
           - prod
 
 jobs:
+  wait-for-checks:
+    runs-on: ubuntu-latest
+    outputs:
+      should_deploy: ${{ steps.check-suite.outputs.should_deploy }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Authenticate GitHub CLI
+        run: echo "${{ github.token }}" | gh auth login --with-token
+
+      - name: Run check-suite script
+        id: check-suite
+        run: |
+          ./.github/workflows/check_suite.sh ${{ github.sha }}
+          echo $GITHUB_OUTPUT
+
   deploy:
     name: Deploy
+    needs: wait-for-checks
+    if: needs.wait-for-checks.outputs.should_deploy == 'true'
```

Comment:
This is not even necessary because I changed it so the script will `exit 1` if it fails and this job depends on that job via `needs`. Now the script will never output `should_deploy = false`. I am going to get rid of it and get rid of all the script outputs.